File: backend/app/models/template.py
from __future__ import annotations

from datetime import datetime

from sqlalchemy import DateTime, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base
from typing import Optional


class Template(Base):
    __tablename__ = "templates"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    platform: Mapped[str] = mapped_column(String(20), nullable=False)
    thumbnail_url: Mapped[Optional[str]] = mapped_column(String(500), nullable=True)
    config: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime, default=datetime.now, nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, default=datetime.now, onupdate=datetime.now, nullable=False
    )

File: backend-go/models/template.go
package models

import (
	"time"
)

type Template struct {
	ID           string    `orm:"column(id);pk;size(36)" json:"id"`
	Name         string    `orm:"column(name);size(200)" json:"name"`
	Platform     string    `orm:"column(platform);size(20)" json:"platform"`
	ThumbnailURL string    `orm:"column(thumbnail_url);size(500);null" json:"thumbnail_url"`
	Config       string    `orm:"column(config);type(text);null" json:"config"`
	CreatedAt    time.Time `orm:"column(created_at);auto_now_add;type(datetime)" json:"created_at"`
	UpdatedAt    time.Time `orm:"column(updated_at);auto_now;type(datetime)" json:"updated_at"`
}

func (t *Template) TableName() string {
	return "templates"
}
